Makefile

```diff
@@ -1,4 +1,146 @@
+LDFLAGS            = -extldflags=-static -w -s
+
+HAS_GOTESTSUM     := $(shell command -v gotestsum;)
+HAS_GOLANGCI_LINT := $(shell command -v golangci-lint;)
+
+.PHONY: help mod-download build install release test coverage lint fmt fmt-test images cloudbuild-test cloudbuild clean install-gotestsum install-golangci-lintall deploy undeploy verify
+
+all: codegen build test lint
+
+# Verify lint and tests
+install-gotestsum:
+ifndef HAS_GOTESTSUM
+	curl -SL https://github.com/gotestyourself/gotestsum/releases/download/v1.7.0/gotestsum_1.7.0_linux_amd64.tar.gz | tar -C $(shell go env GOPATH)/bin -zxf -
+endif
+
+install-golangci-lint:
+ifndef HAS_GOLANGCI_LINT
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell go env GOPATH)/bin v1.41.1
+endif
+
+help: ## Display this help
+	@awk 'BEGIN {FS = ":.*?## "; printf "\n$(KREPO_DESC)\n\nUsage:\n  make \033[36m<cmd>\033[0m\n"} /^[a-zA-Z0-9._-]+:.*?## / {printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+build: $(COMMANDS)  ## Build all artifacts
+
+$(COMMANDS): ## Build artifct
+	CGO_ENABLED=1 $(GO) build -ldflags "$(LDFLAGS)" -o $(BIN_OUTPUT_DIR)/$@ ./cmd/$@
+
+deploy: ## Deploy TriggerMesh stack to default Kubernetes cluster using ko
+	CGO_ENABLED=1 $(KO) deploy -f $(BASE_DIR)/config
+
+undeploy: ## Remove TriggerMesh stack from default Kubernetes cluster using ko
+	$(KO) delete -f $(BASE_DIR)/config
+
+release: ## Build release binaries
+	@set -e ; \
+	for bin in $(COMMANDS) ; do \
+		for platform in $(TARGETS); do \
+			GOOS=$${platform%/*} ; \
+			GOARCH=$${platform#*/} ; \
+			RELEASE_BINARY=$$bin-$${GOOS}-$${GOARCH} ; \
+			[ $${GOOS} = "windows" ] && RELEASE_BINARY=$${RELEASE_BINARY}.exe ; \
+			echo "GOOS=$${GOOS} GOARCH=$${GOARCH} $(GO) build -ldflags "$(LDFLAGS)" -o $(DIST_DIR)/$${RELEASE_BINARY}" ./cmd/$$bin ; \
+			GOOS=$${GOOS} GOARCH=$${GOARCH} $(GO) build -ldflags "$(LDFLAGS)" -o $(DIST_DIR)/$${RELEASE_BINARY} ./cmd/$$bin ; \
+		done ; \
+	done
+
+test: install-gotestsum ## Run unit tests
+	@mkdir -p $(TEST_OUTPUT_DIR)
+	$(GOTEST) -p=1 -race -cover -coverprofile=$(TEST_OUTPUT_DIR)/$(KREPO)-c.out $(GOPKGS)
+
+cover: test ## Generate code coverage
+	@mkdir -p $(COVER_OUTPUT_DIR)
+	$(GOTOOL) cover -html=$(TEST_OUTPUT_DIR)/$(KREPO)-c.out -o $(COVER_OUTPUT_DIR)/$(KREPO)-coverage.html
+
+lint: install-golangci-lint ## Lint source files
+	$(GOLINT) $(GOPKGS)
+
+fmt: ## Format source files
+	$(GOFMT) -s -w $(shell $(GO) list -f '{{$$d := .Dir}}{{range .GoFiles}}{{$$d}}/{{.}} {{end}} {{$$d := .Dir}}{{range .TestGoFiles}}{{$$d}}/{{.}} {{end}}' $(GOPKGS))
+
+fmt-test: ## Check source formatting
+	@test -z $(shell $(GOFMT) -l $(shell $(GO) list -f '{{$$d := .Dir}}{{range .GoFiles}}{{$$d}}/{{.}} {{end}} {{$$d := .Dir}}{{range .TestGoFiles}}{{$$d}}/{{.}} {{end}}' $(GOPKGS)))
+
+verify: lint fmt-test ## Verify code base with lint and fmt
```

Comment:
`lint` includes `gofmt`, so we can probably remove `verify` and `fmt-test` goals.

Reply:
Can remove `verify`, but do think `fmt-test` should stay as a very light weight sanity check on formatting.